dask/utils.py

```diff
@@ -1032,3 +1035,29 @@ def __str__(self):
         return "<%s: %s>" % (self.__class__.__name__, self.token)
 
     __repr__ = __str__
+
+
+def effective_get(get=None, df=None):
+    """Get the effective get method used in a given situation"""
+    df_get = df._default_get if df else None
+    return get or _globals.get('get') or df_get
+
+
+def get_scheduler_lock(get=None, df=None):
+    """Get an instance of the appropriate lock for a certain situation based on
+       scheduler used."""
+    def unwrap_method(f):
+        return getattr(f, '__func__', f)
+
+    actual_get = effective_get(get, df)
+    try:
+        import distributed
+    except ImportError:
+        distributed = None
+
+    if actual_get == multiprocessing.get:
+        return mp.Manager().Lock()
+    if distributed and (unwrap_method(actual_get) is
+                        unwrap_method(distributed.Client.get)):
+        return SerializableLock()
```

Comment:
Dask distributed currently lacks any way to lock writes between processes. This is almost certain to prove problematic eventually. Distributed may eventually need to grow a proper locking service.

Yes, I do think `SerializableLock` is almost strictly better than `threading.Lock`, insofar as it can be pickled.